syntax = "proto3";

import "user.proto";

package Server;

service DatingServer {

  rpc GetUser (UserRequest) returns (UserReply) {}
  rpc SetUser (SetUserRequest) returns (SetUserReply) {}
}

message UserRequest {
  User.TUserKey Key = 1;
}

message UserReply {
  User.TUser User = 1;
}

message SetUserRequest {
  User.TUser User = 1;
}

message SetUserReply {
  User.TUser User = 1;
  string ErrorMessage = 2;
}
